forecasting/src/autogluon/forecasting/predictor.py

```diff
@@ -0,0 +1,420 @@
+            'SFF' (SimpleFeedForward). See References for more detail on these models.
+
+            Values in the `hyperparameters` dict are themselves dictionaries of hyperparameter settings for each model
+            type. Each hyperparameter can either be a single fixed value or a search space containing many possible
+            values. A search space should only be provided when `hyperparameter_tune_kwargs` is specified (i.e.,
+            hyperparameter-tuning is utilized). Any omitted hyperparameters not specified here will be set to default
+            values which are given in`autogluon/forecasting/trainer/models/presets.py`. Specific hyperparameter
+            choices for each of the recommended models can be found in the references.
+        time_limit: int, default=None
+            Approximately how long fit() will run for (wall-clock time in seconds). If not specified, `fit()` will
+            run until all models have completed training.
+
+        Other Parameters
+        ----------------
+        quantile_levels: List[float], default=[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
+            List of increasing decimals that specifies which quantiles should be estimated
+            when making distributional forecasts. Can alternatively be provided with the keyword
+            argument `quantiles`.
+
+        References
+        ----------
+            - DeepAR: https://ts.gluon.ai/api/gluonts/gluonts.model.deepar.html
+            - MQCNN: https://ts.gluon.ai/api/gluonts/gluonts.model.seq2seq.html
+            - SFF: https://ts.gluon.ai/api/gluonts/gluonts.model.simple_feedforward.html
+        """
+        start_time = time.time()
+        if self._learner.is_fit:
+            raise AssertionError(
+                "Predictor is already fit! To fit additional models create a new `Predictor`."
+            )
+
+        verbosity = kwargs.get("verbosity", self.verbosity)
+        set_logger_verbosity(verbosity, logger=logger)
+        if presets is not None:
+            logger.info(f"presets is set to {presets}")
+
+        self.target_column = target_column
+
+        if verbosity >= 3:
+            fit_args = dict(
+                prediction_length=prediction_length,
+                target_column=target_column,
+                hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+                presets=presets,
+                time_limit=time_limit,
+                **kwargs,
+            )
+            logger.info("============ fit arguments ============")
+            logger.info("fit() called with arguments:")
+            logger.info(f"{pprint.pformat(fit_args)}")
+            logger.info("=======================================")
+
+        # Inform the user extra columns in dataset will not be used.
+        extra_columns = [c for c in train_data.columns.copy() if c != target_column]
+        if len(extra_columns) > 0:
+            logger.warning(f"Provided columns {extra_columns} will not be used.")
```

Comment:
I don't see a check on if the target column exists at all in the train data. Perhaps this should be the first thing checked?